From ee4db63cd9b07827649d71b2ada14e256f1c5b70 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 24 Feb 2019 05:09:26 +0100
Subject: [PATCH 1/1] efi_loader: fix entry count in bootmgr

Since commit 914df75b0c97 ("efi_loader: fix EFI entry counting")
entry_count is already set to 1 before efi_bootmgr_load() is called. So we
should not increment it when entering the function.

Without the patch an assert error occurs in efi_get_variable() if DEBUG is
defined.

Fixes: 914df75b0c97 ("efi_loader: fix EFI entry counting")
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_bootmgr.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/efi_loader/efi_bootmgr.c b/lib/efi_loader/efi_bootmgr.c
index 15a06085364..417016102b4 100644
--- a/lib/efi_loader/efi_bootmgr.c
+++ b/lib/efi_loader/efi_bootmgr.c
@@ -174,8 +174,6 @@ void *efi_bootmgr_load(struct efi_device_path **device_path,
 	void *image = NULL;
 	int i, num;
 
-	__efi_entry_check();
-
 	bs = systab.boottime;
 	rs = systab.runtime;
 
@@ -196,7 +194,5 @@ void *efi_bootmgr_load(struct efi_device_path **device_path,
 	free(bootorder);
 
 error:
-	__efi_exit_check();
-
 	return image;
 }
-- 
2.20.1

